arangod/Aql/TraversalExecutor.cpp

```diff
@@ -498,73 +370,21 @@ auto TraversalExecutorInfos::parseTraversalEnumeratorSingleServer(
-
-            _traversalEnumerator =
-                std::make_unique<SingleServerWeightedRefactoredNoneNone>(
-                    SingleServerProvider<SingleServerProviderStep>{
-                        query, std::move(baseProviderOptions),
-                        query.resourceMonitor()},
-                    std::move(enumeratorOptions),
-                    std::move(pathValidatorOptions), query.resourceMonitor());
-            break;
-          case TraverserOptions::UniquenessLevel::PATH:
-          case TraverserOptions::UniquenessLevel::GLOBAL:
-            using SingleServerWeightedRefactoredNonePath =
-                WeightedEnumeratorRefactored<
-                    SingleServerProvider<SingleServerProviderStep>,
-                    VertexUniquenessLevel::NONE, EdgeUniquenessLevel::PATH>;
-
-            _traversalEnumerator =
-                std::make_unique<SingleServerWeightedRefactoredNonePath>(
-                    SingleServerProvider<SingleServerProviderStep>{
-                        query, std::move(baseProviderOptions),
-                        query.resourceMonitor()},
-                    std::move(enumeratorOptions),
-                    std::move(pathValidatorOptions), query.resourceMonitor());
-            break;
-        }
-        break;
-      case TraverserOptions::UniquenessLevel::PATH:
-        using SingleServerWeightedRefactoredPath = WeightedEnumeratorRefactored<
-            SingleServerProvider<SingleServerProviderStep>,
-            VertexUniquenessLevel::PATH, EdgeUniquenessLevel::PATH>;
-
-        _traversalEnumerator =
-            std::make_unique<SingleServerWeightedRefactoredPath>(
-                SingleServerProvider<SingleServerProviderStep>{
-                    query, std::move(baseProviderOptions),
-                    query.resourceMonitor()},
-                std::move(enumeratorOptions), std::move(pathValidatorOptions),
-                query.resourceMonitor());
-        break;
-      case TraverserOptions::UniquenessLevel::GLOBAL:
-        using SingleServerWeightedRefactoredGlobal =
-            WeightedEnumeratorRefactored<
-                SingleServerProvider<SingleServerProviderStep>,
-                VertexUniquenessLevel::GLOBAL, EdgeUniquenessLevel::PATH>;
-
-        _traversalEnumerator =
-            std::make_unique<SingleServerWeightedRefactoredGlobal>(
-                SingleServerProvider<SingleServerProviderStep>{
-                    query, std::move(baseProviderOptions),
-                    query.resourceMonitor()},
-                std::move(enumeratorOptions), std::move(pathValidatorOptions),
-                query.resourceMonitor());
-        break;
-      default:
-        TRI_ASSERT(false);
-        THROW_ARANGO_EXCEPTION(TRI_ERROR_INTERNAL);
-    }
   }
+  // This Assert is not necessary, we could be smart here, for disjoints.
+  // However current implementation does not hand in isSmart==true
+  // in the valid combination.
```

Comment:
I would actually like to keep the Comment.
In case we change something an the Assert triggers, the one hitting it gets the hint that disjoints could be allowed to violate this, if we ever change workflow / parameter propagation there, and does not need to do heavy lifting to figure out how to avoid this assert.